llvm/test/CodeGen/AArch64/GlobalISel/legalize-saddsat.mir

```diff
@@ -220,10 +220,8 @@ body:             |
     ; CHECK-NEXT: [[ICMP:%[0-9]+]]:_(s32) = G_ICMP intpred(ne), [[UADDE]](s32), [[SEXT_INREG2]]
     ; CHECK-NEXT: [[UV4:%[0-9]+]]:_(s8), [[UV5:%[0-9]+]]:_(s8), [[UV6:%[0-9]+]]:_(s8), [[UV7:%[0-9]+]]:_(s8) = G_UNMERGE_VALUES [[UADDE]](s32)
     ; CHECK-NEXT: [[DEF:%[0-9]+]]:_(s8) = G_IMPLICIT_DEF
-    ; CHECK-NEXT: [[DEF1:%[0-9]+]]:_(s32) = G_IMPLICIT_DEF
-    ; CHECK-NEXT: [[UV8:%[0-9]+]]:_(s8), [[UV9:%[0-9]+]]:_(s8), [[UV10:%[0-9]+]]:_(s8), [[UV11:%[0-9]+]]:_(s8) = G_UNMERGE_VALUES [[DEF1]](s32)
     ; CHECK-NEXT: [[MV:%[0-9]+]]:_(s32) = G_MERGE_VALUES [[UV4]](s8), [[UV5]](s8), [[UV6]](s8), [[DEF]](s8)
-    ; CHECK-NEXT: [[MV1:%[0-9]+]]:_(s32) = G_MERGE_VALUES [[UV8]](s8), [[UV9]](s8), [[UV10]](s8), [[UV8]](s8)
+    ; CHECK-NEXT: [[MV1:%[0-9]+]]:_(s32) = G_MERGE_VALUES [[DEF]](s8), [[DEF]](s8), [[DEF]](s8), [[DEF]](s8)
```

Comment:
There are now G_MERGE_VALUES of all undef.

Reply:
Are you saying this is incorrect?

Reply:
No. The long version is that I wanted to highlight a missed optimization opportunity.

Reply:
My other thought was that the merge(unmerge) combine broke.

Reply:
> My other thought was that the merge(unmerge) combine broke.
Where did it break?

Reply:
The G_MERGE_VALUES parameters were all uses of the def's of the G_UNMERGE_VALUES (undef), but it was not a  perfect match.

Reply:
I'm not understanding what exactly was wrong. This diff seems to be correct.

Reply:
It is. I was wondering whether the placement of the combine influences other combines (order matters).